### ISSUE TITLE
How to layer generated code？

### ISSUE THREAD

**Issue by xslasd** [99designs/gqlgen#2731]
### What happened?
When developing a project with multiple people, you want each person to be responsible for one or two modules. I hope Gqlgen can generate architecture code by module.

### What did you expect?

The code architecture is as follows:
```- gqlgen-example
  |- gqlgen.yml    
  |- schema.graphql  
  |- generated  
  |  |- generated.go  
  |  |- models_gen.go  
  |- graphql
  |  |- resolver.go
  |  |- user
  |  |  |- user_resolver_gen.go
  |  |  |- user_models_gen.go
  |  |- article
  |     |- article_resolver_gen.go
  |     |- article_models_gen.go
  |- user.graphql
  |- article.graphql
  |- main.go
```
Can it be achieved through configuration?

**Comment by MrElectroNick** [comment]
Same question

**Comment by x1c1cle** [comment]
+1. our team would like to delegate the development of each resolver to external teams. 
preferably, we would like to have each resolver-directory have its down config file, schema file, and generated code.

**Comment by MrElectroNick** [comment]
Hi @UnAfraid what if allow to follow same approach as with models - make resolvers be configurable via gqlgen.yml like
`resolver:
  UserResolver:
    resolver: [package]/graph/hr_resolvers.UserResolver
`
where `hr_resolver` will be a package

Responsibility for deleted resolvers and circular imports should be developer responsiblility - and planning schemas properly to avoid that also

What do you think?

**Comment by MrElectroNick** [comment]
BTW - it would be also to have an ability to include other yml configs into gqlgen config to be able to separate model definitions or just any large pars of config. In my case I work with project where we already have around 100 types which is a half of the whole DB schema and gqlgen.yml is already 1000 lined

**Comment by x1c1cle** [comment]
> This is technically possible as it is now, the generated code can be split in 2 packages (resolvers/models), but the implementation for each resolver can be separated in their own package.
> 
> Few points about the idea of generating code for each resolver independently:
> 
> * If resolver/model generation was separated in different packages per resolver you may end up with circular imports.
> * If a resolver gets deleted then the previously generated code will most likely be left unused.

hey @UnAfraid thanks for the response. I'm unable to get your suggestion working. can you give some recommendations how to set up the config file to allow isolated resolver generation? thx

**Comment by xslasd** [comment]
@MrElectroNick Can we consider using the generated Go code through registration, similar to GRPC protoc-gen-go?

**Comment by MrElectroNick** [comment]
> @MrElectroNick Can we consider using the generated Go code through registration, similar to GRPC protoc-gen-go?

That's probably a question not addressed to me

**Comment by UnAfraid** [comment]
> > This is technically possible as it is now, the generated code can be split in 2 packages (resolvers/models), but the implementation for each resolver can be separated in their own package.
> > Few points about the idea of generating code for each resolver independently:
> > 
> > * If resolver/model generation was separated in different packages per resolver you may end up with circular imports.
> > * If a resolver gets deleted then the previously generated code will most likely be left unused.
> 
> hey @UnAfraid thanks for the response. I'm unable to get your suggestion working. can you give some recommendations how to set up the config file to allow isolated resolver generation? thx

Here is an example of resolver implementation separation
https://github.com/UnAfraid/wg-ui/blob/master/pkg/api/resolver_root.go

**Comment by MrElectroNick** [comment]
@UnAfraid - thanks for example! It might be really helpfull!

**Comment by x1c1cle** [comment]
thanks @UnAfraid. i see how you got it to work:
- all queries are in a single .graphql file. 
- disable the autogeneration of the resolver.go - so that it can be manually broken out.
- many schema files as needed to provide the necessary isolation.

**Comment by RumenNikiforov** [comment]
> thanks @UnAfraid. i see how you got it to work:
> 
> * all queries are in a single .graphql file.
> * disable the autogeneration of the resolver.go - so that it can be manually broken out.
> * many schema files as needed to provide the necessary isolation.

How you organize your schema doesn't really matter, it was my preference to set it up that way, but you can separate them however you like, the output should still be the same.

**Comment by UnAfraid** [comment]
> thanks @UnAfraid. i see how you got it to work:
> 
> * all queries are in a single .graphql file.
> * disable the autogeneration of the resolver.go - so that it can be manually broken out.
> * many schema files as needed to provide the necessary isolation.

You can organize the schema files in different way, the output should still be the same.

**Comment by x1c1cle** [comment]
> > thanks @UnAfraid. i see how you got it to work:
> > 
> > * all queries are in a single .graphql file.
> > * disable the autogeneration of the resolver.go - so that it can be manually broken out.
> > * many schema files as needed to provide the necessary isolation.
> 
> You can organize the schema files in different way, the output should still be the same, i organized them that way because i prefer to see all the top level queries/mutations/subscriptions in one place, but you can split them out. Even if you use extend type Query/Mutation/Subscription and add them that way, the generated code will essentially be the same.

yup, agreed. There are some limitations, such as: all `type query` must be in the same file. but yea, this definitely will work.

**Comment by UnAfraid** [comment]
> > > thanks @UnAfraid. i see how you got it to work:
> > > 
> > > * all queries are in a single .graphql file.
> > > * disable the autogeneration of the resolver.go - so that it can be manually broken out.
> > > * many schema files as needed to provide the necessary isolation.
> > 
> > 
> > You can organize the schema files in different way, the output should still be the same, i organized them that way because i prefer to see all the top level queries/mutations/subscriptions in one place, but you can split them out. Even if you use extend type Query/Mutation/Subscription and add them that way, the generated code will essentially be the same.
> 
> yup, agreed. There are some limitations, such as: all `type query` must be in the same file. but yea, this definitely will work.

Thats not a limitation really
You can use `extend type Query { users: UsersConnection! }` and `extend type Mutation { createUser(input: CreateUserInput!): CreateUserPayload! }`

**Comment by prusov** [comment]
Exactly the layout we needed for the same team-ownership reason. I built a gqlgen plugin that produces it: https://github.com/prusov/gqldomainresolver

A *domain* is the parent directory of a `.graphqls` file (`graph/schema/user/*.graphqls` → package `user`). Each domain gets its own Go package and never imports `graph/generated`, so domain teams work in isolation and incremental builds stay fast.

Concrete numbers from our codebase: ~121k LOC handwritten Go, ~31k LOC schema, ~123k LOC generated. Cold build went from ~20–30 min to ~50 s after migration. For existing projects there's a step-by-step migration guide ([MIGRATION.md](https://github.com/prusov/gqldomainresolver/blob/master/MIGRATION.md)) — we moved one domain per PR.

Models are still generated centrally (the plugin only restructures resolvers), which for team-ownership is usually what you want anyway.

Would prefer this to be native in gqlgen so the plugin isn't necessary. Disclosure: written with Claude Code; covered by tests.
